File: 3 спринт/I. Conference lovers/Conference_lovers2.py
def Compare(lhs: str, rhs: str):
    if int(lhs) > int(rhs):
        return True
    return False


def GetTop(array, less):
    for i in range(1, len(array)):
        item_to_insert = array[i]
        j = i
        # заменим сравнение item_to_insert < array[j-1] на компаратор less
        while j > 0 and less(Compare, array[j-1]):
            array[j] = array[j-1]
            j -= 1
        array[j] = item_to_insert
        # return [array[a] for a in array]
    for i in range(1, len(array)):
        array[i]
        #и вот здесь остановился 
    return [array[a] for a in array]
    



if __name__ == '__main__':
    q_of_stud = int(input())
    array = input().split()
    result_q_vuz = int(input())
    result = GetTop(q_of_stud, array, result_q_vuz)
    print(result)

File: 3 спринт/I. Conference lovers/Conference_lovers2.cpp
bool Compare(pair <int, int> lhs, pair <int, int> rhs)
    //  Компаратор 
    if (lhs.second == rhs.second)
        return lhs.first < rhs.first
    else
        return lhs.second > rhs.second

array <int> GetTop(array <int> data, int count)
    //  Создаём хеш таблицу и нумеруем её
    hash_table <int, int> counter
    for (val in data)
        counter[val]++
    
    array <pair <int, int>> data == counter
    //  Сортируем с помощью компаратора хеш таблицу
    Sort(data, Compare)

    array <int> result
    // Добавляем в результат количество id университетор по частоте их появления.
    for (int i = 0; i < count; ++i)
        result.Add(data[i])
    return result